modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -401,6 +401,38 @@ void ONNXImporter::populateNet(Net dstNet)
             layerParams.set("pool", layer_type == "GlobalAveragePool" ? "AVE" : "MAX");
             layerParams.set("global_pooling", true);
         }
+        else if (layer_type == "Slice")
+        {
+            if (layerParams.has("axes")) {
+                DictValue axes = layerParams.get("axes");
+                for (int i = 1; i < axes.size(); ++i) {
+                    CV_Assert(axes.get<int>(i - 1) == axes.get<int>(i) - 1);
+                }
+                layerParams.set("axis", axes.get<int>(0));
+            }
+            replaceLayerParam(layerParams, "starts", "begin");
+
+            DictValue ends = layerParams.get("ends");
+            std::vector<int> end(ends.size());
+            for (int i = 0; i < ends.size(); ++i) {
+                int finish = ends.get<int>(i);
+                if (ends.get<int>(i) < 0) {
+                    --finish;
+                }
```

Comment:
If it works like numpy, we need to decrease all the ends because in numpy the ranges are `[begin, end)` but in OpenCV `[begin, end]`.

Reply:
No, if end > 0, then OpenCV range = numpy range

Reply:
And we have no extra information from layer parameters about number of dimensions or something like that?